apps/wallet/src/ui/app/components/accounts/PasswordInputDialog.tsx

```diff
@@ -56,6 +57,28 @@ export function PasswordModalDialog({
         shouldUnregister: true,
     });
 
+    const [countdownError, setCountdownError] = useState<{
+        interval: NodeJS.Timeout;
+        message: string;
+    } | null>(null);
+
+    // Clear the interval and error if the dialog closed but not unmounted
+    useEffect(() => {
+        if (!open && countdownError?.interval) {
+            clearInterval(countdownError.interval);
+            setCountdownError(null);
+        }
+    }, [open]);
+
+    // Clear the interval if the dialog unmounted
+    useEffect(() => {
```

Comment:
These two can't be merge into one `useEffect` ?
Like if you add the return to the first `useEffect` won't it just run when the component unmounts ?